spock-core/src/main/java/org/spockframework/mock/runtime/DynamicProxyMockInterceptorAdapter.java

```diff
@@ -25,6 +27,34 @@ public DynamicProxyMockInterceptorAdapter(IProxyBasedMockInterceptor interceptor
   }
 
   public Object invoke(Object target, Method method, Object[] arguments) throws Throwable {
+    boolean isDefault = (method.getModifiers() & (Modifier.ABSTRACT | Modifier.PUBLIC | Modifier.STATIC)) == Modifier.PUBLIC
```

Comment:
The code for checking if a method is default is actually copied from the implementation of `java.lang.reflect.Method.isDefault()` in Java 8.
The idea here is that the method has to be public, but not abstract and not static. That's why `Modifier.ABSTRACT` and `Modifier.STATIC` must not appear on the right side of the expression.